commons/src/main/java/com/powsybl/commons/datasource/MultipleReadOnlyDataSource.java

```diff
@@ -41,41 +41,41 @@ public String getBaseName() {
     }
 
     @Override
-    public boolean exists(String suffix, String ext) throws IOException {
+    public boolean exists(String suffix, String ext, boolean checkConsistencyWithDataSource) throws IOException {
         return dataSources.stream().anyMatch(dataSource -> {
             try {
-                return dataSource.exists(suffix, ext);
+                return dataSource.exists(suffix, ext, checkConsistencyWithDataSource);
             } catch (IOException e) {
                 throw new UncheckedIOException(e);
             }
         });
     }
 
     @Override
-    public boolean exists(String fileName) throws IOException {
+    public boolean exists(String fileName, boolean checkConsistencyWithDataSource) throws IOException {
         return dataSources.stream().anyMatch(dataSource -> {
             try {
-                return dataSource.exists(fileName);
+                return dataSource.exists(fileName, checkConsistencyWithDataSource);
             } catch (IOException e) {
                 throw new UncheckedIOException(e);
             }
         });
     }
 
     @Override
-    public InputStream newInputStream(String suffix, String ext) throws IOException {
+    public InputStream newInputStream(String suffix, String ext, boolean checkConsistencyWithDataSource) throws IOException {
         for (var dataSource : dataSources) {
-            if (dataSource.exists(suffix, ext)) {
+            if (dataSource.exists(suffix, ext, checkConsistencyWithDataSource)) {
                 return dataSource.newInputStream(suffix, ext);
```

Comment:
should use the same strategy as GenericReadOnlyDataSource. After the exists check, either pass checkConsistencyWithDataSource to newInputStream  or not

Reply:
done